components/contract-components/utils.ts

```diff
@@ -1,8 +1,23 @@
 import { ContractId } from "./types";
-import { KNOWN_CONTRACTS_MAP } from "@thirdweb-dev/sdk";
+import { getErcs } from "@thirdweb-dev/react";
+import { KNOWN_CONTRACTS_MAP, SmartContract } from "@thirdweb-dev/sdk";
 
 export function isContractIdBuiltInContract(
   contractId: ContractId,
 ): contractId is keyof typeof KNOWN_CONTRACTS_MAP {
   return contractId in KNOWN_CONTRACTS_MAP;
 }
+
+export function detectFeature(contract: SmartContract | null, feature: string) {
+  const { erc721, erc1155, erc20 } = getErcs(contract);
+
+  if (!contract || (!erc721 && !erc1155 && !erc20)) {
+    return false;
+  }
+
+  return (
+    (erc721 && feature in erc721) ||
+    (erc1155 && feature in erc1155) ||
+    (erc20 && feature in erc20)
```

Comment:
This won't work. The property will always be on the object now, but calling it might throw an exception.

I think you should use the actual `isFeatureEnabled()` function instead

Reply:
did a wrapping of it to check multiple features